services/submission/src/main/java/app/coronawarn/server/services/submission/normalization/SubmissionKeyNormalizer.java

```diff
@@ -0,0 +1,60 @@
+/*-
+ * ---license-start
+ * Corona-Warn-App
+ * ---
+ * Copyright (C) 2020 SAP SE and all other contributors
+ * ---
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ---license-end
+ */
+
+package app.coronawarn.server.services.submission.normalization;
+
+import app.coronawarn.server.common.persistence.domain.normalization.DiagnosisKeyNormalizer;
+import app.coronawarn.server.common.persistence.domain.normalization.NormalizableFields;
+import app.coronawarn.server.services.submission.config.SubmissionServiceConfig;
+import java.util.Map;
+
+public final class SubmissionKeyNormalizer implements DiagnosisKeyNormalizer {
+
+  private final Map<Integer, Integer> dsosFromTrlMap;
+
+  public SubmissionKeyNormalizer(SubmissionServiceConfig config) {
+    dsosFromTrlMap = config.getTekFieldDerivations().getDsosFromTrl();
+  }
+
+  @Override
+  public NormalizableFields normalize(NormalizableFields fieldsAndValues) {
+    int trlValue = fieldsAndValues.getTransmissionRiskLevel();
+    int dsos = fieldsAndValues.getDaysSinceOnsetOfSymptoms();
+
+    throwIfAllRequiredFieldsMissing(trlValue, dsos);
+
+    if (isMissing(dsos)) {
+      dsos = dsosFromTrlMap.getOrDefault(trlValue, 0);
```

Comment:
Wouldn't this set `dsos` to negative values? Since the mapping in `application.yaml` is from 0 to -4. I wonder if this is fine, since the guard check above `isMissing` validates that values are greater than 0.

Reply:
That's a good point. I think the check for missing field should strictly verify the 0 value.

Reply:
On checking the TEK protobuf class again , i found that we can verify whether an optional field was provided or not, so i will change the implementation to better test this situatio. Initially i thought these integer fields are initialized with 0 in case not provided.

Reply:
sounds good